pkg/fakedata/template.go

```diff
@@ -0,0 +1,171 @@
+	},
+	"ProductName": func() string {
+		return generators["product.name"].Func(Column{Name: "tmplProductName", Key: "tmplProductNameKey", Constraints: ""})
+	},
+	"EventAction": func() string {
+		return generators["event.action"].Func(Column{Name: "tmplEventAction", Key: "tmplEventActionKey", Constraints: ""})
+	},
+	"HTTPMethod": func() string {
+		return generators["http.method"].Func(Column{Name: "tmplHTTPMethod", Key: "tmplHTTPMethodKey", Constraints: ""})
+	},
+	"Name": func() string {
+		return generators["name"].Func(Column{Name: "tmplName", Key: "tmplNameKey", Constraints: " "})
+	},
+	"Email": func() string {
+		return generators["email"].Func(Column{Name: "tmplEmail", Key: "tmplEmailKey", Constraints: " "})
+	},
+	"Domain": func() string {
+		return generators["domain"].Func(Column{Name: "tmplDomain", Key: "tmplDomainKey", Constraints: ""})
+	},
+	"IPv4": func() string {
+		return generators["ipv4"].Func(Column{Name: "tmplIPv4", Key: "tmplIPv4Key", Constraints: ""})
+	},
+	"IPv6": func() string {
+		return generators["ipv6"].Func(Column{Name: "tmplIPv6", Key: "tmplIPv6Key", Constraints: ""})
+	},
+	"MacAddress": func() string {
+		return generators["mac.address"].Func(Column{Name: "tmplMacAddress", Key: "tmplMacAddressKey", Constraints: ""})
+	},
+	"Latitude": func() string {
+		return generators["latitude"].Func(Column{Name: "tmplLatitude", Key: "tmplLatitudeKey", Constraints: ""})
+	},
+	"Longitude": func() string {
+		return generators["longitude"].Func(Column{Name: "tmplLongitude", Key: "tmplLongitudeKey", Constraints: ""})
+	},
+	"Double": func() string {
+		return generators["double"].Func(Column{Name: "tmplDouble", Key: "tmplDoubleKey", Constraints: ""})
+	},
+	"Int": func(params ...int) string {
+		constraint := ".."
+		a := 0
+		b := 1000
+		if len(params) == 1 {
+			a = params[0]
+		}
+		if len(params) == 2 {
+			a = params[0]
+			b = params[1]
+		}
+		constraint = fmt.Sprintf("%d..%d", a, b)
+		return generators["int"].Func(Column{"Int", "int", constraint})
+	},
+	"Enum": func(keywords ...string) string {
+		constraints := createConstraints(keywords)
+		return generators["enum"].Func(Column{"Enum", "enu", constraints})
+	},
+	"File": func(path string) string {
+		return generators["file"].Func(Column{Name: "tmplA", Key: "tmplAKey", Constraints: path})
+	},
+}
+
+func createConstraints(params []string) string {
+	return strings.Join(params, "..")
+}
+
+func getTemplateNameFromPath(name string) string {
+	ts := strings.FieldsFunc(name, splitPathName)
+	tn := ts[len(ts)-1]
+	return tn
+}
+
+// this custom split function is used with strings.FieldsFunc to split the path
+// by `/` (Unix, MacOS) or `\` (Windows) for absolute and relative paths to template files
+func splitPathName(r rune) bool {
+	return r == '/' || r == '\\'
+}
+
+// ParseTemplate takes a path to a template file as argument. It parses the template file and executes it on
+// os.Stdout.
+func ParseTemplate(path string, limit int) (err error) {
```

Comment:
@lucapette do you have an idea how we could test this function? It needs a file on disk. Should we provide a file and if so, where would we place it? Together with the golden files for the integration tests?